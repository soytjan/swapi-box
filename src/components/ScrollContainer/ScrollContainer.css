.fade {
  position: relative;
  width: 100%;
  min-height: 30vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1; }

.scroll-container {
  display: flex;
  justify-content: center;
  height: 500px;
  perspective: 300px;
  color: #feda4a;
  font-size: 1.5em;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify; }

.scroll-text {
  position: relative;
  top: -50px;
  transform-origin: 50% 100%;
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 60s linear; }

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px); } }
