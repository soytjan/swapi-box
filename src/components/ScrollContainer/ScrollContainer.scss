.fade {
  position: relative;
  width: 100%;
  min-height: 30vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.scroll-container {
  display: flex;
  justify-content: center;
  /* This is a magic number based on the context in which this snippet is used and effects the perspective */
  height: 500px;
  /* This sets allows us to transform the text on a 3D plane, and is somewhat a magic number */
  perspective: 300px;
  color: #feda4a;
  font-size: 2em;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.scroll-text {
  position: relative;
  top: -50px;
  /* Defines the skew origin at the very center when we apply transforms on the animation */
  transform-origin: 50% 100%;
  position: relative;
  /* Defines the skew origin at the very center when we apply transforms on the animation */
  transform-origin: 50% 100%;
  /* Adds the crawl animation, which plays for one minute */
  animation: crawl 60s linear;
}

@keyframes crawl {
  0% {
    /* The element starts below the screen */
    top: 0;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    /* Slightly increasing the rotation at the end and moving the text far away from the viewer */
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
