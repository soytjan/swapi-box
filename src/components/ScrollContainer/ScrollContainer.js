import React from 'react';
import PropTypes from 'prop-types';
import './ScrollContainer.css';

const ScrollContainer = ({title, crawl, releaseDate }) => {
  return (
    <div>
      <div className="fade"></div>
      <div className='scroll-container'>
        <div className='scroll-text'> 
          <p>
            {crawl}
          </p>
          <h3>{title}</h3>
          <h4>{releaseDate}</h4>
        </div>
      </div>
    </div>
  );
};

ScrollContainer.propTypes = {
  title: PropTypes.string,
  crawl: PropTypes.string,
  releaseDate: PropTypes.string
};

export default ScrollContainer;