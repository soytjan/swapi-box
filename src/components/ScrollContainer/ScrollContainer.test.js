import React from 'react';
import ScrollContainer from './ScrollContainer';

describe('ScrollContainer', () => {
  it('expect to exist', () => {
    expect(ScrollContainer).toBeDefined();
  })
})