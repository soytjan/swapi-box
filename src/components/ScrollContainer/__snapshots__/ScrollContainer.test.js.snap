// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollContainer
    crawl="War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor...."
    releaseDate="2005-05-19"
    title="Revenge of the Sith"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <div
          className="scroll-container"
>
          <div
                    className="scroll-text"
          >
                    <p>
                              War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
                    </p>
                    <h3>
                              Revenge of the Sith
                    </h3>
                    <h4>
                              2005-05-19
                    </h4>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="scroll-text"
>
            <p>
                        War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
            </p>
            <h3>
                        Revenge of the Sith
            </h3>
            <h4>
                        2005-05-19
            </h4>
</div>,
          "className": "scroll-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
</p>,
              <h3>
                Revenge of the Sith
</h3>,
              <h4>
                2005-05-19
</h4>,
            ],
            "className": "scroll-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....",
              },
              "ref": null,
              "rendered": "War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Revenge of the Sith",
              },
              "ref": null,
              "rendered": "Revenge of the Sith",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2005-05-19",
              },
              "ref": null,
              "rendered": "2005-05-19",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <div
            className="scroll-container"
>
            <div
                        className="scroll-text"
            >
                        <p>
                                    War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
                        </p>
                        <h3>
                                    Revenge of the Sith
                        </h3>
                        <h4>
                                    2005-05-19
                        </h4>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="scroll-text"
>
              <p>
                            War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
              </p>
              <h3>
                            Revenge of the Sith
              </h3>
              <h4>
                            2005-05-19
              </h4>
</div>,
            "className": "scroll-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....
</p>,
                <h3>
                  Revenge of the Sith
</h3>,
                <h4>
                  2005-05-19
</h4>,
              ],
              "className": "scroll-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....",
                },
                "ref": null,
                "rendered": "War! The Republic is crumbling under attacks by t…ate mission to rescue the captive Chancellor....",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Revenge of the Sith",
                },
                "ref": null,
                "rendered": "Revenge of the Sith",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2005-05-19",
                },
                "ref": null,
                "rendered": "2005-05-19",
                "type": "h4",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
